pliers/extractors/text.py

```diff
@@ -313,3 +332,114 @@ def _extract(self, stim):
         features = ['sentiment_' + k for k in scores.keys()]
         return ExtractorResult([list(scores.values())], stim, self,
                                features=features)
+
+
+class SpaCyTokenExtractor(TextExtractor):
+    '''
+    Uses SpaCy to extract features from text. Extracts features for every word(token) in a sentence.
+    Features returned are:
+     text : Verbatim text content.(unicode)
+     lemma_ : Base form of the token, with no inflectional suffixes.(unicode)
+     pos_ : Coarse-grained part-of-speech.(unicode)
+     tag_ : Fine-grained part-of-speech.(unicode)
+     dep_ : Syntactic dependency relation.(unicode)
+     shape_ : Transform of the tokens's string, to show orthographic features. For example, "Xxxx" or "dd".(unicode)
+     is_alpha : Does the token consist of alphabetic characters? (bool)
+     is_stop : Is the token part of a "stop list"? (bool)
+     is_punct : Is the token punctuation?(bool)
+     sentiment : A scalar value indicating the positivity or negativity of the token.(float)
+     is_ascii : Does the token consist of ASCII characters?(bool)
+     is_digit : Does the token consist of digits?(bool)
+    '''
+
+    def __init__(self, model='en_core_web_sm', features=None):
+        verify_dependencies(['spacy'])
+        try:
+            self.model = spacy.load(model)
+
+            print('loaded model: ', self.model)
+        except (ImportError, IOError, OSError) as e:
+            logging.error(
+                "Spacy Models ('{ "+model+" }') not found. Downloading and installing")
+
+            os.system('python -m spacy download '+model)
+            model_na = model[:2]
+            self.model = spacy.load(model_na)
```

Comment:
I'm assuming the reason you're taking just the language part of the model name is to try and load a related model if the specific one the user asked for can't be found. But we probably don't want to do that in this case, because then the user won't know that we loaded a different model.

I.e., suppose they pass `model='en_new_fancy_model'`. That doesn't actually exist, but because spacy will successfully load *something* if we try to load just `'en'`, we're now in a position where we no longer know exactly which model was loaded. We could in theory track this internally, but it's probably easier and safer to just fail. So let's change this to just `load(model)`, and if it fails, the user will know that they're passing an invalid model name.